canonical-web-and-design/maas-ui
# feat(ui): controller configuration tab

## Done

- controller configuration tab
- added "note" field to the controller configuration (it's supported and wasn't implemented in the legacy version)
- refactor `DeviceConfigurationFields` to `NodeConfigurationFields` to support controllers
-  refactor `PowerParameters` to work for controllers    
-  update `controller/types/actions.ts` with valid supported parameters

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to controller details configuration and verify all options work as expected (various power options in particular)

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/946
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/947

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/7452681/171864171-68940b31-f988-4e60-931c-2db3b5967d77.png">
